### ISSUE TITLE
Physics: Coplanar waveguide (CPWG) impedance calculator

### ISSUE THREAD

**Issue by rjwalters** [rjwalters/kicad-tools#351]
## Overview

Implement coplanar waveguide with ground (CPWG) impedance calculations. CPWG is common for RF designs and provides better isolation than microstrip.

## Parent Issue

Part of #348 Physics Module: Transmission Line Calculator

### CPWG Calculator

```python
# Addition to src/kicad_tools/physics/transmission_line.py

class TransmissionLine:
    # ... existing methods ...
    
    def cpwg(
        self,
        width_mm: float,
        gap_mm: float,
        layer: str,
        frequency_ghz: float = 1.0,
    ) -> ImpedanceResult:
        """
        Calculate coplanar waveguide with ground impedance.
        
        CPWG geometry:
        
            gap   width   gap
        ═══════╗ ═══════ ╔═══════  ← Signal trace with coplanar ground
               ╚═════════╝
        ─────────────────────────  ← Ground plane below
        
        Args:
            width_mm: Center conductor width
            gap_mm: Gap between conductor and coplanar ground
            layer: Layer name
            frequency_ghz: Frequency for loss calculation
            
        Returns:
            ImpedanceResult with Z0, εeff, loss, velocity
        """
        h = self.stackup.get_reference_plane_distance(layer)
        er = self.stackup.get_dielectric_above(layer).dielectric_constant
        t = self.stackup.get_copper_thickness(layer)
        
        return self._cpwg_calc(width_mm, gap_mm, h, er, t, frequency_ghz)
    
    def cpwg_geometry_for_impedance(
        self,
        z0_target: float,
        layer: str,
        width_mm: float | None = None,
        gap_mm: float | None = None,
    ) -> tuple[float, float]:
        """
        Calculate CPWG geometry for target impedance.
        
        Provide either width or gap; the other is calculated.
        If neither provided, optimizes for minimum area.
        
        Returns:
            (width_mm, gap_mm) tuple
        """
```

### CPWG Equations

```python
def _cpwg_calc(
    self,
    w: float,
    g: float,
    h: float,
    er: float,
    t: float,
    freq: float,
) -> ImpedanceResult:
    """
    CPWG impedance using conformal mapping.
    
    Based on Ghione & Naldi analysis.
    """
    import math
    from scipy.special import ellipk  # Complete elliptic integral
    
    # Geometry ratios
    a = w / 2
    b = w / 2 + g
    
    # Elliptic integral arguments
    k0 = a / b
    k0_prime = math.sqrt(1 - k0**2)
    
    k1 = math.sinh(math.pi * a / (2 * h)) / math.sinh(math.pi * b / (2 * h))
    k1_prime = math.sqrt(1 - k1**2)
    
    # Effective dielectric constant
    q = (ellipk(k1) * ellipk(k0_prime)) / (ellipk(k1_prime) * ellipk(k0))
    eps_eff = 1 + (er - 1) * q / 2
    
    # Characteristic impedance
    z0 = (60 * math.pi / math.sqrt(eps_eff)) * (1 / (ellipk(k0) / ellipk(k0_prime) + ellipk(k1) / ellipk(k1_prime)))
    
    # Phase velocity
    c = 299792458
    v_p = c / math.sqrt(eps_eff)
    
    # Loss estimation
    loss = self._estimate_cpwg_loss(w, g, h, er, t, freq)
    
    return ImpedanceResult(
        z0=z0,
        epsilon_eff=eps_eff,
        loss_db_per_m=loss,
        phase_velocity=v_p,
    )
```

## Files to Modify

- `src/kicad_tools/physics/transmission_line.py` - Add CPWG methods

## Dependencies

- scipy (for elliptic integrals) - already available via numpy ecosystem

## Acceptance Criteria

- [ ] CPWG Z0 calculation ±3% vs reference
- [ ] Handle finite ground plane width
- [ ] Inverse calculation (specify width OR gap)
- [ ] Loss estimation for CPWG
- [ ] Unit tests against known values

## Test Cases

```python
def test_cpwg_50ohm():
    """Standard 50Ω CPWG on FR4."""
    stackup = Stackup.jlcpcb_4layer()
    tl = TransmissionLine(stackup)
    
    # Typical 50Ω CPWG: ~10mil trace, ~6mil gap
    result = tl.cpwg(width_mm=0.25, gap_mm=0.15, layer="F.Cu")
    assert result.z0 == pytest.approx(50, rel=0.1)
```

## References

- Ghione & Naldi, "Analytical Formulas for Coplanar Lines in Hybrid and Monolithic MICs"
- IPC-2141A Section 4.4

## Related

- #348 Physics Module (parent)
- #349 Stackup parsing (dependency)
- #350 Microstrip/stripline (sibling)
